File: clean-code/Python/5/5_1.py
def analyze_data(data):
    if data and len(data) > 0:
        total = 0
        for value in data:
            if value > 0:
                if value % 2 == 0:
                    total += value
                else:
                    total -= value
            else:
                print("Non-positive value encountered:", value)
        print("Final total:", total)
    else:
        print("Data is empty or undefined.")

File: clean-code/C#/5/5_1.cs
void AnalyzeData(List<int> data)
{
    if (data != null && data.Count > 0)
    {
        int total = 0;
        foreach (int value in data)
        {
            if (value > 0)
            {
                if (value % 2 == 0)
                {
                    total += value;
                }
                else
                {
                    total -= value;
                }
            }
            else
            {
                Console.WriteLine("Non-positive value encountered: " + value);
            }
        }
        Console.WriteLine("Final total: " + total);
    }
    else
    {
        Console.WriteLine("Data is empty or undefined.");
    }
}
